prog/dftb+/lib_reks/reksinterface.F90

```diff
@@ -292,7 +292,9 @@ subroutine getReksGradients(env, denseDesc, sccCalc, rangeSep, dispersion, &
 
     end if
 
-    call setReksGradients_(derivs, self)
+    if (allocated(self%SSRgrad)) then
+      call setReksGradients_(derivs, self)
+    end if
```

Comment:
self%SSRgrad is always allocated when we calculate the gradient. Why is the condition `allocated(self%SSTgrad)` added?

Reply:
Its not always allocated, the HBI+H2O_periodic_stress case, for example, tries to access this variable without it being allocated with some compilers. Possibly its an intent() issue somewhere, but it returns the correct forces with if test in place.

Reply:
The Thymine_single case is easier to test (much faster to run), and for the intel compiler with the debug setting, in main.F90 allocated(reks%SSRgrad) is false at all of the places I have tested it.

Reply:
I modified this line, please see commit 6dc8ae2, the `setReksGradients_` routine moved to inner part of if condition (`Efunction /= 1`) so that it is not needed to access this routine for single-state REKS.